core/src/main/java/tc/oc/pgm/destroyable/DestroyableFactory.java

```diff
@@ -33,15 +56,17 @@ public DestroyableFactory(
       double destructionRequired,
       @Nullable ImmutableSet<Mode> modeList,
       boolean showProgress,
-      boolean sparks,
-      boolean repairable) {
+      String sparks,
+      boolean repairable,
+      Element el)
```

Comment:
As mentioned in another comment, don't pass the element here, do the parsing outside

```suggestion
      SparksType sparks,
      boolean repairable)
```